### PR TITLE
visualization: Add anatomy of icicle graphs and how to interpret them

### PR DESCRIPTION
**Pull request by kakkoyun**
Add two document pages to explain icicle graphs and flame graphs.

It uses https://rive.app to create animations.

#### References:
- https://rive.app/learn-rive
- https://help.rive.app/runtimes/overview/react
- https://github.com/rive-app/rive-react
- https://help.rive.app/runtimes/overview/react/parameters-and-return-values
- https://help.rive.app/runtimes/layout

### CODE REVIEW

File: docs/icicle-graph-anatomy.mdx [hunk 1]
```diff
@@ -0,0 +1,329 @@
...
+This is how we visualize a single call stack collected from a program.
+But happens when we collect multiple call stacks over time?
+
+Let's see.
+
+## Anatomy of Continuous profiling
+
+In the life-cycle of a program, there are many call stacks.
+Even in a second, a program can have thousands of calls and in a profile that we collected from a program for a minute,
+there can be millions of call stacks. And we you profile a host with multiple programs running, there can be billions of call stacks.
```
Comment (yomete) [inline docs/icicle-graph-anatomy.mdx:224]:
```suggestion
there can be millions of call stacks. And when you profile a host with multiple programs running, there can be billions of call stacks.
```

**Review by yomete — APPROVED** [review]
Very awesome additions to the docs Kemal! 👏🏾

**Comment by yomete** [comment]
One other thing, as this has happened about 2-3 times while I was checking this out locally. The documentation page `Anatomy of icicle and flame graphs` sometimes crashes after spending a period of time on it. This happens on Google Chrome for me. Curious to know if this happens to any other person.
Screenshot 2023-03-23 at 19 53 38

I imagine this could be because of the multiple animations on the page, and we can solve that by lazy loading those animations so that they're only loaded when they are visible in the browser window. Again, this could just me over exerting my computer's resources and its a non-issue.

**Comment by kakkoyun** [comment]
> One other thing, as this has happened about 2-3 times while I was checking this out locally. The documentation page `Anatomy of icicle and flame graphs` sometimes crashes after spending a period of time on it. This happens on Google Chrome for me. Curious to know if this happens to any other person. Screenshot 2023-03-23 at 19 53 38
> 
> I imagine this could be because of the multiple animations on the page, and we can solve that by lazy loading those animations so that they're only loaded when they are visible in the browser window. Again, this could just me over exerting my computer's resources and its a non-issue.

I've never encountered this. Thanks for letting me know. I'll try to figure out how to lazy load Rive animations.

**Comment by yomete** [comment]
> > One other thing, as this has happened about 2-3 times while I was checking this out locally. The documentation page `Anatomy of icicle and flame graphs` sometimes crashes after spending a period of time on it. This happens on Google Chrome for me. Curious to know if this happens to any other person. Screenshot 2023-03-23 at 19 53 38
> > I imagine this could be because of the multiple animations on the page, and we can solve that by lazy loading those animations so that they're only loaded when they are visible in the browser window. Again, this could just me over exerting my computer's resources and its a non-issue.
> 
> I've never encountered this. Thanks for letting me know. I'll try to figure out how to lazy load Rive animations.

Yeah, like i said, it could also just be my computer and it's a non-issue. Also happy to help with lazy loading the animations. Let me know if you'd like that.

**Pull request merged**